src/liballoc/lib.rs

```diff
@@ -78,7 +78,6 @@
 #[macro_use]
 extern crate core;
 
-#[cfg(all(not(feature = "external_funcs"), not(feature = "external_crate")))]
```

Comment:
That removes the error but links to libc instead. Adding `#[allow(unused_features)]` is a way better 'fix' (it still allows it globally...) without introducing a pointless platform specific dependency. IMHO the real problem is that it isn't possible to add `#![feature(libc)]` only when no other features are present...